docs: weekly architecture docs refresh

enyst opened the pull request:
Summary of updates

This weekly refresh updates the SDK architecture documentation under sdk/arch based on the latest HEAD of OpenHands/software-agent-sdk (commit 93d405c9).

Updated pages

1) sdk/arch/agent-server.mdx
- Rewrote the page to accurately reflect the current FastAPI app, routers, services, and WebSocket endpoints.
- Clarified that the Agent Server itself does not orchestrate containers; containerization is handled by workspace implementations (e.g., DockerWorkspace) that launch the server.
- Added explicit endpoint inventory grouped by router, and authentication details (X-Session-API-Key header for HTTP and session_api_key query parameter for WebSockets).
- Included component diagrams aligned with current code.
- Added Last updated timestamp and source commit.

2) sdk/arch/security.mdx
- Fixed incorrect source link path (openhands-sdk/openhands/sdk/security).
- Added ConfirmationPolicy reference to the key components table.
- Appended Last updated timestamp and source commit.

Citations (inline in the docs)
- openhands-agent-server/openhands/agent_server/api.py (app wiring, lifespan)
- openhands-agent-server/openhands/agent_server/dependencies.py (authentication headers and query parameter for WebSocket)
- openhands-agent-server/openhands/agent_server/conversation_router.py (conversation endpoints)
- openhands-agent-server/openhands/agent_server/event_router.py (event endpoints)
- openhands-agent-server/openhands/agent_server/sockets.py (WebSocket endpoints)
- openhands-agent-server/openhands/agent_server/file_router.py (file upload/download)
- openhands-agent-server/openhands/agent_server/git_router.py (git endpoints)
- openhands-agent-server/openhands/agent_server/tool_router.py (tool listing)
- openhands-workspace/openhands/workspace/docker/workspace.py (server launched by workspace)
- openhands-sdk/openhands/sdk/security/* (analyzers, risks, confirmation policy)

Source of truth
- software-agent-sdk commit: 93d405c91edbaccd84d7b0a356340123a911a6cb

Process notes
- Followed the Guide for Writing Architecture Component Documentation.
- Kept changes scoped to architecture pages; no guides or API reference modified beyond necessary link fixes.
- CI: Please run pre-commit hooks in CI as configured.


@enyst can click here to [continue refining the PR](https://app.all-hands.dev/conversations/cf2a8e922e384dcdad33e7f9fe2c7583)

enyst committed:
docs: weekly architecture docs refresh

- Update agent-server.mdx to reflect current server responsibilities and endpoints (source: openhands-agent-server/*)
- Fix security.mdx source link; add ConfirmationPolicy reference; add Last updated and Source commit footer

Co-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
docs: add inline GitHub source links per guidelines (agent-server.mdx)\n\nCo-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
docs: move Last updated footer to end of file (security.mdx)\n\nCo-authored-by: openhands <openhands@all-hands.dev>

xingyaoww reviewed (commented):
Weird -- i don't think agent server package is included in docs.json?

enyst committed:
docs(agent-server): add Client SDK section with REST + WebSocket examples, inline source links\n\nCo-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
docs(agent-server): add bash-events WebSocket streaming example with inline source links\n\nCo-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
docs(nav): add SDK architecture link for agent-server to left nav

Place agent-server at the end of the SDK Components section for consistent ordering.

Co-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
docs(agent-server): resolve conflicts, fix Mermaid blocks, rename to 'Agent Server', and simplify selected endpoints

Also keeps concise Client SDK REST + WebSocket examples with inline GitHub source links.

Co-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
Merge branch 'main' into docs-weekly-arch-refresh

enyst committed:
sdk/arch(agent-server): fix Mermaid syntax for Mintlify (escape wildcards, avoid /* and braces in labels)

- Replace '/api/*' and '/sockets/*' with safe labels in diagrams
- Replace '{id}' with '<id>' in node label

Co-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
docs(nav): move Agent Server architecture page under Remote Agent Server group\n\n- Remove sdk/arch/agent-server from SDK Components\n- Add sdk/arch/agent-server under Guides > Remote Agent Server (after overview)\n\nCo-authored-by: openhands <openhands@all-hands.dev>

xingyaoww committed:
Merge branch 'main' into docs-weekly-arch-refresh

xingyaoww reviewed (commented):
The more i read into this doc, the more i feel it probably make sense to combine this docs with agent-server overview page we currently have, and have this doc link to specific docs of Docker/APIWorkspace when discussing about *Workspace that spawns agent server, wdyt?

enyst commented:
> The more i read into this doc, the more i feel it probably make sense to combine this docs with agent-server overview page we currently have, and have this doc link to specific docs of Docker/APIWorkspace when discussing about *Workspace that spawns agent server, wdyt?

Yes I feel maybe there’s some duplication. @openhands review the pages under agent-server and use your taste to combine meaningfully; link to workspaces.

enyst committed:
docs: consolidate agent-server architecture and guides pages

- Refactored sdk/arch/agent-server.mdx to focus on technical architecture:
  - Removed duplicate content and client SDK examples
  - Added clear API endpoints table
  - Added authentication table
  - Added links to workspace implementations (Docker, API, Cloud)
  - Added See Also section linking to guides

- Refactored sdk/guides/agent-server/overview.mdx to focus on practical usage:
  - Renamed title to 'Remote Agent Server' for clarity
  - Removed duplicate architecture diagrams
  - Added workspace types comparison table
  - Added OpenHands Cloud workspace example
  - Simplified 'How It Works' section
  - Added links to architecture page for technical details

- Updated Next Steps sections in all agent-server guide pages for consistency

Co-authored-by: openhands <openhands@all-hands.dev>

enyst committed:
Merge branch 'main' into docs-weekly-arch-refresh

enyst commented on docs.json:
```diff
@@ -245,6 +245,8 @@
                 "group": "Remote Agent Server",
                 "pages": [
                   "sdk/guides/agent-server/overview",
+                      "sdk/arch/agent-server",
```
@openhands I think we already have in the `overview` page similar content with the `agent-server` page? Verify what I just said, and dedup: remove from agent-server page what we already basically cover in overview, and move from agent-server page to overview page the rest. Don't spend much explanations on workspaces, link to the respective workspace instead.

Do make sure to keep all contents about the seamless transition from local to remote, and of course the rest of the non-dupe content.

enyst committed:
docs: consolidate agent-server content into overview

- Move unique content from sdk/arch/agent-server.mdx to sdk/arch/overview.mdx
- Remove sdk/arch/agent-server.mdx (content now in overview)
- Update all references to point to the new location
- Update docs.json navigation to remove agent-server page

Co-authored-by: openhands <openhands@all-hands.dev>
